python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -184,39 +195,41 @@ def generate_thresholded_overlap_scores_dosage(
     gwas_scores_path: str, dosage_matrix_path: str, p_value_threshold: float
 ) -> pd.DataFrame:
     """Compare and restrict to overlapping loci between dosage matrix and thresholded scores."""
-    scores = _preprocess_scores(gwas_scores_path)
-    dosage_loci_nomiss = _extract_nomiss_dosage_loci(dosage_matrix_path)
+    thresholded_scores_chunks = []
 
-    thresholded_indices_set = set()
-    for chunk in read_feather_in_chunks(gwas_scores_path, columns=["P"], chunk_size=1000):
+    for chunk in read_feather_in_chunks(gwas_scores_path, chunk_size=1000):
         thresholded_indices = get_p_value_thresholded_indices(chunk, p_value_threshold)
-        thresholded_indices_set.update(thresholded_indices)
-    scores_filtered_indices = sorted(list(thresholded_indices_set))
-    thresholded_scores = scores.iloc[scores_filtered_indices]
-    thresholded_scores = thresholded_scores.set_index("SNPID")
+        if thresholded_indices:
+            thresholded_scores_chunks.append(chunk.loc[thresholded_indices])
 
-    if len(thresholded_scores) == 0:
+    if not thresholded_scores_chunks:
         raise ValueError("No thresholded scores available; cannot proceed with PRS calculation.")
 
-    set_A = set(thresholded_scores.index)
-    set_B = set(dosage_loci_nomiss["locus"])
+    thresholded_scores = pd.concat(thresholded_scores_chunks)
+    preprocessed_thresholded_scores = _preprocess_scores(thresholded_scores)
+    preprocessed_thresholded_scores = preprocessed_thresholded_scores.set_index("SNPID")
+    thresholded_score_loci = set(preprocessed_thresholded_scores.index)
+    dosage_loci_nomiss = _extract_nomiss_dosage_loci(dosage_matrix_path, thresholded_score_loci)
+    filtered_dosage_loci = set(dosage_loci_nomiss["locus"])
 
-    overlap_snps = set_A.intersection(set_B)
-    remaining_snps_scores = len(thresholded_scores) - len(overlap_snps)
-    if remaining_snps_scores == 0:
+    overlap_loci = thresholded_score_loci.intersection(filtered_dosage_loci)
+    remaining_loci = len(preprocessed_thresholded_scores) - len(overlap_loci)
```

Comment:
Is the purpose of this check to make sure we have some overlap? If so, it doesn't work when there is perfect overlap. Do we want to check `len(overlap_loci) > 0` instead?